### PR TITLE
Fix broken PSD estimation workflow

### PR DESCRIPTION
**Pull request by titodalcanton**
PSD estimation workflow is currently broken by some of the recent changes to the segment APIs. This patch gets it to a state where it works and produces results again. However the segment functions are not completely straightforward to me. I tried to apply the same changes that I think went into `pycbc_make_coinc_search_workflow`. Ian/Alex, could you take a look and see if you spot some obviously wrong usage?

There are two additional issues that I want to address next (parallelization of calculate_psd jobs and size of summary page) but I'll deal with those in separate commits.

### CODE REVIEW

**Comment by spxiwh** [comment]
Hi @titodalcanton, I have a couple of concerns about this (namely SCIENCE_OK is not what should be analysed, [SCIENCE_OK - DATA_NOT_PRESENT] should be used), but probably easier to briefly discuss this over a coffee when I am Hannover tomorrow and übertomorrow.

(P.S. I am officially making übertomorrow an English word).

**Comment by titodalcanton** [comment]
Great, let's überdiscuss tomorrow then!

**Comment by spxiwh** [comment]
Discussed this with @titodalcanton yesterday. We agreed on changes; once these are made I will be happy to merge this.

**Comment by spxiwh** [comment]
Any update on this Tito? Assigning this back to you for now. Please assign back to me when updated.

**Comment by titodalcanton** [comment]
I just pushed more changes that address your points and also split the single calculate_psd jobs into several ones. This gets the workflow to complete in a reasonable time, an example is here:

https://www.atlas.aei.uni-hannover.de/~tito/LSC/o1/o1-psd-estim-c02-test/

Although the PSD looks ok, there are remaining things to fix in the summary page (namely add workflow link on the top and fix the segment plot at the end of the page). But if you're ok with the status so far, feel free to merge and I'll do the remaining fixes in a new PR.

File: bin/hdfcoinc/pycbc_make_psd_estimation_workflow [hunk 1]
```diff
@@ -71,74 +71,98 @@ save_fig_with_metadata(time_str, time_file.storage_path, **kwds)
...
-        pycbc.workflow.get_cumulative_veto_group_files(
-                workflow, 'segments-veto-groups', 'segments')
-final_veto_file, final_veto_name, ind_cats = \
-        pycbc.workflow.get_cumulative_veto_group_files(
-                workflow, 'segments-final-veto-group', "segments")
+seg_dir = 'segments'
+
+veto_cat_files = pycbc.workflow.get_files_for_vetoes(
+        workflow, seg_dir, runtime_names=['segments-science-veto'],
+        in_workflow_names=['segments-final-veto-group', 'segments-veto-groups'])
```
Comment (spxiwh) [inline bin/hdfcoinc/pycbc_make_psd_estimation_workflow]: Do you need "in_workflow_names" here? This implies you are adding jobs to the workflow to run (I assume) CAT 2,3,4 veto generation. I think you only need ['segments-science-veto'] (ie. what is removed in get_analysable_segments). So I think in_workflow_names=None (which is the default value) should be used.

**Comment by spxiwh** [comment]
Assigning to @titodalcanton to address comment.

**Comment by titodalcanton** [comment]
Thanks Ian, the change seems to work (the workflow is generated correctly) although Atlas has not been very cooperative in running the test workflow.

**Pull request merged**